server/pbench/bin/index-pbench

```diff
@@ -187,99 +319,303 @@ def es_index(es, actions, dbg=0):
+            'metric': 'util',
+            'display': 'Utilization',
+            'units': 'percent',
+            'subfields': [],
+            'colpat': re.compile(r'(?P<id>.+)')
+        },
+        'disk_Wait_Time_msec.csv': {
+            'class': 'disk',
+            'metric': 'wtime',
+            'display': 'Wait_Time',
+            'units': 'msec',
+            'subfields': [ 'read', 'write' ],
+            'colpat': re.compile(r'(?P<id>.+)-(?P<subfield>read|write)')
+        }
+    },
+    'proc-interrupts': {
+        '@prospectus': {
+            # For proc-interrupts .csv files, we want to individually index
+            # each column entry as its own JSON document
+            'handling': 'individual'
+        },
+    },
+    'sar': {},
+    'turbostat': {}
+}
+# If we need to deal with old .csv file names, place an alias here to map to
+# an existing handler above.
+_aliases = {
+    'disk_Request_Merges.csv': 'disk_Request_Merges_per_sec.csv',
+    'disk_Request_Size.csv': 'disk_Request_Size_in_512_byte_sectors.csv',
+    'disk_Throughput.csv': 'disk_Throughput_MB_per_sec.csv',
+    'disk_Utilization.csv': 'disk_Utilization_percent.csv',
+    'disk_Wait_Time.csv': 'disk_Wait_Time_msec.csv',
+}
 
-def mk_run_metadata(mdconf, md5sum, dirname, dirprefix):
-    try:
-        d = {}
-        d.update(mdconf.items('run'))
-        d.update(mdconf.items('pbench'))
-        # Fix up old format dates so ES won't barf on them.
-        d['start_run'] = d['start_run'].replace('_', 'T')
-        d['end_run'] = d['end_run'].replace('_', 'T')
-        # Convert the date to UTC: date and start-run are supposed to be close
-        # (except that date may be in local time and start-run is UTC).
-        d['date'] = _to_utc(d['date'].replace('_', 'T'), d['start_run'])
-        # the run id here is the md5sum that's the _id of the main document.
-        # it's what ties all of the relevant documents together.
-        d['id'] = md5sum
-        d['tarball-dirname'] = dirname
-        d['tarball-toc-prefix'] = dirprefix
-        del d['rpm-version']
-        return d
-    except KeyError as e:
-        print("Tarball: %s - %s is missing in metadata.log\n" % (dirname, e), file=sys.stdout)
-        return {}
 
-def mk_user_specified_metadata(options):
-    # parse the JSON string into a dict and return it
-    try:
-        return json.loads(options.metadata_string)
-    except Exception:
-        print("Cannot parse JSON string: {}\n".format(options.metadata_string))
-        return {}
+class ToolData(object):
+    def __init__(self, tstamp, md5sum, experiment, iteration, sample, host, tool, toolgroup,
+                 mdlog, tb):
+        self.toolname = tool
+        (iterseqno, itername) = iteration.split('-', 1)
+        self.run_metadata = {
+            "runtstamp": tstamp,
+            "runid": md5sum,
+            "experiment": experiment,
+            # FIXME: What are the constituent parts of an iteration name?
```

Comment:
From Nick, the parts of the iteration name are per-benchmark, and are today the elements of the various for loops that drive the iterations.  We may need to document this for each benchmark in table like the tool data does.